Assets/Scripts/Door.cs

```diff
@@ -11,14 +11,27 @@ public class Door : MonoBehaviour
     public EventReference doorCloseEvent;
     public EventReference doorCreakLoopEvent;
 
-    public bool isHeld = false;
+    private bool isHeld = false;
     public bool isLocked = false;
 
     private Coroutine playSoundCoroutine = null;
     private EventInstance doorCreakInstance;
 
+    private HingeJoint hinge;
+    private JointLimits limits;
+
     public float velocityThreshold = 0.1f;
-    public float stopDelayDuration = 0.5f;
+    public float stopDelayDuration = 2f;
+
+    [SerializeField]
+    private float initialAngle;
+
+    private void Start()
+    {
+        hinge = GetComponent<HingeJoint>();
```

Comment:
Assuming that each door has a hinge, you should just add requirescomponent above the class so it auto adds a hinge to it then.